### ISSUE TITLE
[vSphere] Calico Service CIDR Changes Causes Kube-DNS Provision Error 

### ISSUE THREAD

**Issue by mwatermolen** [rancher/rke#2740]
### TLDR;
Creating a cluster w/ Calico and a modified service CIDR on vSphere (Possibly others) w/ rancher 2.5.8 Causes a Kube-DNS provision error loop

![image](https://user-images.githubusercontent.com/73605619/141191309-1b7fe2a5-fc27-4322-aa8a-f1fb339489f6.png)

### docker version

```
Client:
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.8
 Git commit:        20.10.7-0ubuntu5~20.04.2
 Built:             Mon Nov  1 00:34:17 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       20.10.7-0ubuntu5~20.04.2
  Built:            Fri Oct 22 00:45:53 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.5.5-0ubuntu3~20.04.1
  GitCommit:
 runc:
  Version:          1.0.1-0ubuntu2~20.04.1
  GitCommit:
 docker-init:
  Version:          0.19.0
  GitCommit:
```

### docker info
```
Client:
 Context:    default
 Debug Mode: false

Server:
 Containers: 32
  Running: 11
  Paused: 0
  Stopped: 21
 Images: 9
 Server Version: 20.10.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2 io.containerd.runtime.v1.linux
 Default Runtime: runc
 Init Binary: docker-init
 containerd version:
 runc version:
 init version:
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.4.0-90-generic
 Operating System: Ubuntu 20.04.3 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 3.843GiB
 Name: lnk-rke-cnppx-cp-3
 ID: O27A:6GXK:AHWL:S3Y4:FWHK:CITX:IBDK:CRW7:VWEP:3J5V:PCYJ:QNUU
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
  provider=vmwarevsphere
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Default Address Pools:
   Base: 172.220.0.0/16, Size: 24

WARNING: No swap limit support
```

### UNAME -R
```
5.4.0-90-generic
```
### cat /etc/os-release
```
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

## **cluster.yml file:**
```yaml
answers: {}
docker_root_dir: /var/lib/docker
enable_cluster_alerting: false
enable_cluster_monitoring: false
enable_network_policy: false
fleet_workspace_name: fleet-default
local_cluster_auth_endpoint:
  enabled: true
name: lnk-rke-cnppx
rancher_kubernetes_engine_config:
  addon_job_timeout: 45
  authentication:
    strategy: x509|webhook
  authorization: {}
  bastion_host:
    ssh_agent_auth: false
  cloud_provider: {}
  dns:
    linear_autoscaler_params:
      cores_per_replica: 128
      max: 0
      min: 1
      nodes_per_replica: 4
      prevent_single_point_failure: true
    node_selector: null
    nodelocal:
      node_selector: null
      update_strategy:
        rolling_update: {}
    options: null
    reversecidrs: null
    stubdomains: null
    tolerations: null
    update_strategy:
      rolling_update: {}
    upstreamnameservers: null
  ignore_docker_version: true
  ingress:
    default_backend: false
    http_port: 0
    https_port: 0
    provider: nginx
  kubernetes_version: v1.20.11-rancher1-1
  monitoring:
    provider: metrics-server
    replicas: 1
  network:
    mtu: 0
    options:
      flannel_backend_type: vxlan
    plugin: calico
  restore:
    restore: false
  rotate_encryption_key: false
  services:
    etcd:
      backup_config:
        enabled: true
        interval_hours: 12
        retention: 6
        safe_timestamp: false
        timeout: 300
      creation: 12h
      extra_args:
        election-timeout: '5000'
        heartbeat-interval: '500'
      gid: 0
      retention: 72h
      snapshot: false
      uid: 0
    kube-api:
      always_pull_images: false
      pod_security_policy: false
      secrets_encryption_config:
        enabled: false
      service_cluster_ip_range: 10.51.0.0/16
      service_node_port_range: 30000-32767
    kube-controller:
      cluster_cidr: 10.50.0.0/16
      service_cluster_ip_range: 10.51.0.0/16
    kubelet:
      fail_swap_on: false
      generate_serving_certificate: false
    kubeproxy: {}
    scheduler: {}
  ssh_agent_auth: false
  upgrade_strategy:
    drain: false
    max_unavailable_controlplane: '1'
    max_unavailable_worker: 10%
    node_drain_input:
      delete_local_data: false
      force: false
      grace_period: -1
      ignore_daemon_sets: true
      timeout: 120
scheduled_cluster_scan:
  enabled: false
  scan_config:
    cis_scan_config:
      override_benchmark_version: rke-cis-1.5
      profile: permissive
  schedule_config:
    cron_schedule: 0 0 * * *
    retention: 24

```

## **Steps to Reproduce:** 
1. Provision a vCenter Cluster with this yaml
2. Wait and then Sadness

## **Results:**

This 
```
[addons] Executing deploy job rke-coredns-addon
```

followed by this in a loop
```
Failed to get job complete status for job rke-coredns-addon-deploy-job in namespace kube-system
```

## Additional Info
VM Templates are prepared with Packer using this script

```bash
# Apply updates and cleanup Apt cache
# packer build --var-file=variables.json ubuntu-2004.json
apt-get update ; apt-get -y dist-upgrade
apt-get -y autoremove
apt-get -y clean
apt-get install docker.io open-iscsi curl net-tools nfs-kernel-server uuid -y

# Disable swap - generally recommended for K8s, but otherwise enable it for other workloads
echo "Disabling Swap"
swapoff -a
sed -i '/ swap / s/^\(.*\)$/#\1/g' /etc/fstab

# Reset the machine-id value. This has known to cause issues with DHCP
#
echo "Reset Machine-ID"
truncate -s 0 /etc/machine-id
rm /var/lib/dbus/machine-id
ln -s /etc/machine-id /var/lib/dbus/machine-id

# Reset any existing cloud-init state
#
echo "Reset Cloud-Init"
rm /etc/cloud/cloud.cfg.d/*.cfg
cloud-init clean -s -l

# Prevent cloud-init from setting IP
#
echo "Disabling cloud-init networking"
bash -c "echo 'network: {config: disabled}' > /etc/cloud/cloud.cfg.d/99-disable-network-config.cfg"

# Installing Automox
curl -sS "https://console.automox.com/downloadInstaller?accesskey=<REDACTED>" | sudo bash

# Harden Kubernetes
chown etcd:etcd /var/lib/etcd

# Rancher be happy
useradd -g docker -m docker 

cat <<'EOF' >>/etc/docker/daemon.json
{
 "default-address-pools":
 [
 {"base":"172.220.0.0/16","size":24}
 ]
}
EOF



useradd -g sudo -s /bin/bash -m <REDACTED>
useradd -g sudo -s /bin/bash -m <REDACTED>

sudo -u <REDACTED>ssh-import-id-gh <REDACTED>
sudo -u <REDACTED>ssh-import-id-gh <REDACTED>

# passwd -d <REDACTED>
# passwd -d <REDACTED>


# zabbix Jazz
wget https://repo.zabbix.com/zabbix/5.4/ubuntu/pool/main/z/zabbix-release/zabbix-release_5.4-1+ubuntu$(lsb_release -rs)_all.deb -O zabbix.deb
dpkg -i zabbix.deb
apt update
apt install zabbix-agent
echo "<REDACTED>" > /etc/zabbix/zabbix.psk



df -h
```

**Comment by mwatermolen** [comment]
Welp after I rolled the kubelet cert it fell into place after 2 hours. 

I did verify with a 2nd identical cluster that this fix was repeatable

**HOWEVER** the cattle cluster agent is tring to reach DNS at the default 10.43.0.10, we also seem to be missing a lot of default services


![image](https://user-images.githubusercontent.com/73605619/141192345-5f2d7961-aa49-4f75-90a4-9b1c46d7c693.png)

**Comment by superseb** [comment]
If you change this, it needs `cluster_dns_server` on the `kubelet` service. (see https://rancher.com/docs/rke/latest/en/example-yamls/)

I can run a test if it properly works on my end.

**Comment by superseb** [comment]
Tested successfully using this YAML:

```
services:
  kube-api:
    service_cluster_ip_range: 10.51.0.0/16
  kube-controller:
    cluster_cidr: 10.50.0.0/16
    service_cluster_ip_range: 10.51.0.0/16
  kubelet:
    cluster_dns_server: 10.51.0.10

network:
  plugin: calico
```

**Comment by mwatermolen** [comment]
@superseb That was the issue! Thanks :) Might be worth adding to https://rancher.com/docs/rke/latest/en/config-options/services/ ;)
